### PR TITLE
ROX-19729: Add label selector for secured cluster

### PR DESCRIPTION
**Pull request by kurlov**
## Description

Add an ability for secured cluster reconciler to reconcile only labeled CRs

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Locally tested with 
```
make install run
```

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: operator/pkg/common/extensions/mapkubeapis.go [hunk 1]
```diff
@@ -80,3 +81,19 @@ func (r *mapKubeAPIsExtensionRun) Execute() error {
...
 }
+
+// AddSelectorOptionIfNeeded conditionally adds label selector to opts for reconciler
+func AddSelectorOptionIfNeeded(selector string, opts []pkgReconciler.Option) ([]pkgReconciler.Option, error) {
+	if len(selector) == 0 {
+		return opts, nil
+	}
+	labelSelector, err := v1.ParseToLabelSelector(selector)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse central label selector")
```
Comment (ludydoo) [inline operator/pkg/common/extensions/mapkubeapis.go:92]:
```suggestion
		return nil, errors.Wrap(err, "failed to parse selector")
```

File: operator/pkg/common/extensions/mapkubeapis.go [hunk 2]
```diff
@@ -80,3 +81,19 @@ func (r *mapKubeAPIsExtensionRun) Execute() error {
 
 	return nil
 }
+
+// AddSelectorOptionIfNeeded conditionally adds label selector to opts for reconciler
+func AddSelectorOptionIfNeeded(selector string, opts []pkgReconciler.Option) ([]pkgReconciler.Option, error) {
```
Comment (ludydoo) [inline operator/pkg/common/extensions/mapkubeapis.go:86]: This should be in another file, not `mapkubeapis`

Comment (kurlov) [reply operator/pkg/common/extensions/mapkubeapis.go:86]: I will add a separate file for it. Not sure it belongs to any existing one

**Review by ludydoo — APPROVED** [review]

**Review by SimonBaeumer — APPROVED** [review]

**Pull request merged**